### ISSUE TITLE
feature: ValidationError to extend native Error class

### ISSUE THREAD

**Issue by GabCostaSilva** [typestack/class-validator#1715]
### Description
Using class validator I'm having issues to properly test an entity custom validation with proper assertions because Jest can't catch the validation error because it expects an Error, and ValidationError doesn’t extend any native error class.

### Proposed solution
Just extend the native Error class in ValidationError.

**Comment by NoNameProvided** [comment]
Generally, I am not against extending the `ValidationError` from the `Error` object, but I think this is not a straightforward change, and just extending it without other changes is no better than the current situation. 

The reasons why I am saying it:
- Currently, we don't throw the `ValidationError` we simply return it that's why it is a class. Returning real errors without throwing is kind of anti-pattern in my book.
- Errors have a `name` property that will be repeated for every error instance we return. Since we may validate huge objects that will just take up space in the logs, etc, and make debugging harder, maybe even confusing for some.
- The `stack` property will point to an invalid place. Where we raised the error now is where the problem is. This again will be confusing for some. 


To sum up, I am not against this change, but when we change this we should also change the API to throw on validation errors instead of returning them. However, this introduces other tasks:
- as first step decide if it's better than the current approach (I think so, but we need discussion)
- create new error(s) to represent internal errors in the library
- performance check, throwing instead of returning may be significantly slower, so we need to measure before making the change

As you can see this is not a simple one-line change. 

I will leave the issue open for further discussion.
